Lesson 2 - Functions/code/exercise1/code.js

```diff
@@ -9,9 +9,19 @@
         var sum = 0;
         
         // !!! add all arguments to sum
+        for(var i = 0; i < arguments.length; i++) {
+            sum += arguments[i];
+        }
         
         // !!! return function for use sum value as a base for next arythmetic operation but without change value of sum variable
         return function() {
+            var result = sum;
```

Comment:
See `array.reduce` for more efficient and shorter version. This is more functional approach. :)